packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -95,15 +95,23 @@ export const createPluginActions: CreatePluginActions = (
 
   return {
     actions: {
-      setContent: (content) => {
+      setContent: (action) => {
+        const content: unknown =
+          // This is not safe: if the user pass a function to setContent(),
```

Comment:
Let's say you want to save a function as content 

```js
setContent((arg, arg2) => 'hello')
```

Then the `setContent` implementation above will invoke that function, and your content will be set to 'hello'.

The workaround would be to call

```js
setContent(() => (arg1, arg2) => 'hello')
```

(of course, you can serialize a function to json, so it's going to crash)

I tried to type it safer, but the problem is that any union type with `unknown` will resolve to `unknown`

```ts
// Equivalent types
type A = unknown | (value: hello) => void 
type B =  unknown
```

This is the same way React behaves though.

Reply:
Ah 🤔 but even if user intends to set a function as content, Storyblok cannot save (serialize) the function inside the story. So we shouldn't have to care about this case, no?

Reply:
@eunjae-lee It's just that we'd ideally have an error thrown in this case, rather than invoking the function. As you say, I think it should be okay. It's just not so ideal to have to override typescript like this 🙂

Reply:
@johannes-lindgren Oh now I get what you meant by that comment 👍🏼